lightly_studio/src/lightly_studio/core/dataset.py

```diff
@@ -288,6 +303,36 @@ def add_samples_from_path(
             image_paths=image_paths,
         )
 
+        if tag_depth == 1 and created_sample_ids:
+            print(f"Adding directory tags to {len(created_sample_ids)} new samples.")
+            parent_dir_to_sample_ids = defaultdict(list)
+
+            # Fetch newly created samples to get their file paths
+            newly_created_samples = (
+                self.query()
+                .match(SampleField.id.is_in(created_sample_ids))
+                .to_list()
+            )
+
+            for sample in newly_created_samples:
+                parent_dir = Path(sample.file_path_abs).parent.name.strip() 
+                if not parent_dir:
+                    parent_dir = "root"  
+                parent_dir_to_sample_ids[parent_dir].append(sample.sample_id)
```

Comment:
The current implementation is contrary to the GitHub issue https://github.com/lightly-ai/lightly-studio/issues/7#issuecomment-3465939257, which tells to not set a tag when an image is in the root directory.

```suggestion
                if parent_dir:
                    parent_dir_to_sample_ids[parent_dir].append(sample.sample_id)
```